x-pack/plugins/maps/public/classes/styles/vector/components/legend/style_error.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { EuiCallOut, EuiFlexGroup, EuiFlexItem, EuiText, EuiToolTip } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { DynamicStyleProperty } from '../../properties/dynamic_style_property';
+import { DataRequest } from '../../../../util/data_request';
+
+interface Props {
+  error: Error;
+  style: DynamicStyleProperty<DataRequest>;
+}
+
+export const StyleError = ({ error, style }: Props) => {
+  const [label, setLabel] = useState('');
+  const styleName = style.getDisplayStyleName();
+
+  useEffect(() => {
+    const getLabel = async () => {
+      const field = style.getField();
+      if (!field) {
+        return;
+      }
+      const fieldLabel = await field.getLabel();
```

Comment:
Any time a useEffect function awaits, you need a flag to abort the results if the useEffect fires again, before the previous function finishes executing.

```
useEffect(() => {
    let canceled = false;
    const getLabel = async () => {
      const field = style.getField();
      if (!field) {
        return;
      }
      const fieldLabel = await field.getLabel();
      if (canceled) {
        return;
      }
      setLabel(fieldLabel);
    };

    getLabel();

    return () => {
      canceled = true;
    }
  }, [style]);
```

Reply:
Oops, yeah I forgot about this. Thanks for pointing this out.